### DISCUSSION TITLE
Bottlerocket docker default LOGGING DRIVER

### DISCUSSION THREAD

**Discussion by mbaciu-gpsw** [bottlerocket-os/bottlerocket#1929] [Q&A]
Hi there,

Context: Bottlerocket instances in AWS ECS clusters.

I was wondering if there's a way to configure the default docker logging driver. This seems to be set to `journald` now (I'm looking at an instance started from the latest image).
I'm trying to change it to `json-file`.

What I tried:
1. putting
```
[settings.ecs]
cluster = "cluster-name"
logging-drivers = ["json-file"]
```
in user data

2. using `apiclient set` to alter `settings.ecs.logging-drivers`

None of the above seem to impact the docker daemon running config.
I know I can always change the `task definition` `logging-driver` part of ECS services to `json-file` (now it's empty, i.e. "use default driver"), but that is suboptimal for a large number of services, so I don't want to go there yet.

Thanks!

**Comment by jpculp** [comment]
Thanks for reaching out, @mbaciu-gpsw! The `logging-drivers` setting is just a list of drivers available for ECS placement decisions, but does not change anything on the host. Can you elaborate further as to your use case for making `json-file` the default?

**Comment by mbaciu-gpsw** [comment]
Hi @jpculp ,

my use case relates to gradually shifting from Amazon Linux to Bottlerecket instances for ECS workloads (EKS will follow). We'd like to make the transition as smooth as possible, with as little changes to workloads as possible.
Now, for container log collection we rely on the datadog agent, which requires all containers to output logs as files in json format.
Historically, we've handled this by setting the default docker logging driver to `json-file` and leaving the logging driver section of the container definition empty (which causes them to rely on the docker daemon defaults).
Having the option to change the default logging driver will help speed unblock the transition to bottlerocket. If that's not possible next best is to change the task definitions of running services, but given the number of running services it might not prove feasible.

Thanks for your help and please let me know if there's any other information I can provide.

**Comment by jpculp** [comment]
@mbaciu-gpsw, thanks for the follow-up! We're going to have to look into this a little bit more.

**Reply by jpculp** [reply]
I've gone ahead and opened issue #1951 so this doesn't get lost.
